packages/core-js/utils/hardhat/argument-types.js

```diff
@@ -16,6 +16,37 @@ const alphanumeric = {
   },
 };
 
+const address = {
+  name: 'address',
+  parse: (argName, value) => value,
+  validate: (argName, value) => {
+    const valid = typeof value === 'string' && /^0x[a-fA-F0-9]{40}$/.test(value);
+    if (!valid) {
+      throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
+        value,
+        name: argName,
+        type: address.name,
+      });
+    }
+  },
+};
+
+const oneOf = (...values) => ({
+  name: 'oneOf',
+  parse: (argName, value) => value,
+  validate: (argName, value) => {
+    if (!values.includes(value)) {
+      throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
+        value,
+        name: argName,
+        type: `oneOf(${values.join('|')})`,
+      });
+    }
+  },
+});
+
 module.exports = {
   alphanumeric,
+  address,
+  oneOf,
```

Comment:
Added some nice hardhat type arguments to be able to remote copy pasting from the tasks.